internal/controller/datadogagent/controller_reconcile_v2_common_test.go

```diff
@@ -215,9 +226,589 @@ func Test_shouldCheckCreateStrategyStatus(t *testing.T) {
+				ExtendedDaemonsetOptions: agent.ExtendedDaemonsetOptions{
+					Enabled: true,
+				},
+				DatadogAgentProfileEnabled: true,
+			},
+			profile:           &testProfile,
+			ddaLastUpdateTime: now,
+			now:               now,
+			existingObjects: []client.Object{
+				&edsdatadoghqv1alpha1.ExtendedDaemonSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-eds",
+						Namespace: testNS,
+						Labels:    testEDSLabels,
+					},
+				},
+				&edsdatadoghqv1alpha1.ExtendedDaemonSetReplicaSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-ers-1",
+						Namespace: testNS,
+						Labels:    testERSLabels,
+					},
+				},
+				&edsdatadoghqv1alpha1.ExtendedDaemonSetReplicaSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-ers-2",
+						Namespace: testNS,
+						Labels:    testERSLabels,
+					},
+				},
+			},
+			expectedShouldUpdate: false,
+			errorMessage:         nil,
+		},
+		{
+			name: "ERS name and active replicaSet name don't match",
+			reconcilerOptions: ReconcilerOptions{
+				ExtendedDaemonsetOptions: agent.ExtendedDaemonsetOptions{
+					Enabled: true,
+				},
+				DatadogAgentProfileEnabled: true,
+			},
+			profile:           &testProfile,
+			ddaLastUpdateTime: now,
+			now:               now,
+			existingObjects: []client.Object{
+				&edsdatadoghqv1alpha1.ExtendedDaemonSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-eds",
+						Namespace: testNS,
+						Labels:    testEDSLabels,
+					},
+					Status: edsdatadoghqv1alpha1.ExtendedDaemonSetStatus{
+						ActiveReplicaSet: "",
+					},
+				},
+				&edsdatadoghqv1alpha1.ExtendedDaemonSetReplicaSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-ers-1",
+						Namespace: testNS,
+						Labels:    testERSLabels,
+					},
+				},
+			},
+			expectedShouldUpdate: false,
+			errorMessage:         errors.New("ExtendedDaemonSetReplicaSet name does not match ExtendedDaemonSet's active replicaset"),
+		},
+		{
+			name: "DDA was just updated",
+			reconcilerOptions: ReconcilerOptions{
+				ExtendedDaemonsetOptions: agent.ExtendedDaemonsetOptions{
+					Enabled: true,
+				},
+				DatadogAgentProfileEnabled: true,
+			},
+			profile:           &testProfile,
+			ddaLastUpdateTime: now,
+			now:               now,
+			existingObjects: []client.Object{
+				&edsdatadoghqv1alpha1.ExtendedDaemonSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-eds",
+						Namespace: testNS,
+						Labels:    testEDSLabels,
+					},
+					Status: edsdatadoghqv1alpha1.ExtendedDaemonSetStatus{
+						ActiveReplicaSet: "test-ers-1",
+					},
+				},
+				&edsdatadoghqv1alpha1.ExtendedDaemonSetReplicaSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-ers-1",
+						Namespace: testNS,
+						Labels:    testERSLabels,
+					},
+				},
+			},
+			expectedShouldUpdate: false,
+			errorMessage:         nil,
+		},
+		{
+			name: "Canary validated annotation present but name doesn't match",
```

Comment:
I don't understand this specific test, what does it validate ? I understand we should update instantly if the valid ERS is `test-ers-2` while the active ERS is `test-ers-1`, but I don't understand the scenario under which it happens: a manual canary validation ?

Reply:
🤔 I might have mixed myself up when I made some edits. I'll switch it to the validation annotation being present but the agent spec hashes not matching since that can happen if a canary is manually validated and then the DDA is updated again but the canary fails